pennylane/resources/df.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for resource estimation with double factorization method.
+"""
+
+from pennylane import numpy as np
+
+
+def norm(one, two, eigvals):
+    r"""Return the 1-norm of a molecular Hamiltonian from the one- and two-electron integrals and
+    eigenvalues of the factorized two-electron integral tensor.
+
+    The 1-norm of a double-factorized molecular Hamiltonian is computed as
+    [`arXiv:2007.14460 <https://arxiv.org/abs/2007.14460>`_]
+
+    .. math::
+
+        \lambda = ||T|| + \frac{1}{4} \sum_r ||L^{(r)}||^2,
+
+    where the Schatten norm for a given matrix :math:`T` is defined as
+
+    .. math::
+
+        ||T|| = \sum_k |\text{eigvals}[T]_k|.
+
+    The matrices :math:`L^{(r)}` are obtained from factorization of the two-electron integral
+    tensor :math:`V` such that
+
+    .. math::
+
+        V_{ijkl} = \sum_r L_{ij}^{(r)} L_{kl}^{(r) T}.
+
+    The matrix :math:`T` is constructed from the one- and two-electron integrals as
+
+    .. math::
+
+        T = h_{ij} - \frac{1}{2} \sum_l V_{illj} + \sum_l V_{llij}.
+
+    Note that the two-electron integral tensor must be arranged in the chemist notation.
+
+    Args:
+        one (array[array[float]]): one-electron integrals
+        two (array[array[float]]): two-electron integrals
+        eigvals (array[float]): eigenvalues of the matrices obtained from factorizing the
```

Comment:
Is it really required that we pass the eigenvalues? Can't they just be absorbed inside the integrals?

My understanding of the double factorization is we express V in terms of L's, and the L's are further truncated. But still, we have an equation for V and therefore also the full Hamiltonian. So why can't we compute the norm more directly?